packages/patternfly-4/react-core/src/components/StyledSystem/StyledBox.js

```diff
@@ -0,0 +1,285 @@
+import React from 'react';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import styled from 'react-emotion';
+import { ThemeProvider } from 'emotion-theming';
+import {
+  space,
+  color,
+  display,
+  maxWidth,
+  minWidth,
+  width,
+  height,
+  maxHeight,
+  minHeight,
+  verticalAlign,
+  flex,
+  order,
+  flexBasis,
+  alignSelf,
+  fontSize,
+  borders,
+  borderColor,
+  borderRadius,
+  boxShadow,
+  opacity,
+  overflow,
+  ratio,
+  background,
+  backgroundImage,
+  backgroundSize,
+  backgroundPosition,
+  backgroundRepeat,
+  position,
+  zIndex,
+  left,
+  top,
+  right,
+  bottom
+} from 'styled-system';
+import { StyledTheme } from './StyledTheme';
+
+export const StyledBoxBase = styled('div')(
+  // theme key: space
+  space,
+  // theme key: maxWidths
+  maxWidth,
+  // theme key: minWidths
+  minWidth,
+  // theme key: heights
+  height,
+  // theme key: maxHeights
+  maxHeight,
+  // theme key: minHeights
+  minHeight,
+  // theme key: fontSizes
+  fontSize,
+  // theme key: colors
+  color,
+  // theme key: borders
+  borders,
+  // theme key: colors
+  borderColor,
+  // theme key: radii
+  borderRadius,
+  // theme key: shadows
+  boxShadow,
+
+  // no theme keys
+  display,
+  width,
+  verticalAlign,
+  opacity,
+  overflow,
+  ratio,
+
+  // flex child props
+  flex,
+  flexBasis,
+  order,
+  alignSelf,
+
+  // background props
+  background,
+  backgroundImage,
+  backgroundSize,
+  backgroundPosition,
+  backgroundRepeat,
+
+  // position props
+  position,
+  zIndex,
+  left,
+  top,
+  right,
+  bottom
+);
+
+const StyledBox = ({ className, children, component, ...props }) => {
+  const StyledBoxBaseWithComponent = StyledBoxBase.withComponent(component);
+  return (
+    <ThemeProvider theme={StyledTheme}>
```

Comment:
Wonder if we should take the time to add a `PatternFlyThemeProvider` that contains this. That would allow people to overwrite the theme anywhere in the tree. Also, it would take some of the overhead out of this.